autoload/asyncomplete.vim

```diff
@@ -10,141 +16,87 @@ if !has('timers')
+            else
+                let s:on_change_manager = l:change_manager()
+            endif
+            if has_key(s:on_change_manager, 'error')
+                call asyncomplete#log('core', 'initializing asyncomplete change manager failed', s:on_change_manager['name'], s:on_change_manager['error'])
+            else
+                call s:on_change_manager.register(function('s:on_change'))
+                call asyncomplete#log('core', 'initializing asyncomplete change manager complete', s:on_change_manager['name'])
+                break
+            endif
+        endfor
 
-    let b:asyncomplete_enable = 1
-    if exists('##TextChangedP')
-        augroup asyncomplete
-            autocmd! * <buffer>
-            autocmd InsertEnter <buffer> call s:remote_insert_enter()
-            autocmd InsertLeave <buffer> call s:remote_insert_leave()
-            autocmd TextChangedI <buffer> call s:on_changed()
-            autocmd TextChangedP <buffer> call s:on_changed_p()
-            autocmd FileType <buffer> call s:file_type_changed()
-        augroup END
-    else
         augroup asyncomplete
-            autocmd! * <buffer>
-            autocmd InsertEnter <buffer> call s:remote_insert_enter()
-            autocmd InsertLeave <buffer> call s:remote_insert_leave()
-            autocmd InsertEnter <buffer> call s:change_tick_start()
-            autocmd InsertLeave <buffer> call s:change_tick_stop()
-            autocmd FileType <buffer> call s:file_type_changed()
-            " working together with timer, the timer is for detecting changes
-            " popup menu is visible. TextChangedI will not be triggered when popup
-            " menu is visible, but TextChangedI is more efficient and faster than
-            " timer when popup menu is not visible.
-            autocmd TextChangedI <buffer> call s:check_changes()
+            autocmd!
+            autocmd InsertEnter * call s:on_insert_enter()
+            autocmd InsertLeave * call s:on_insert_leave()
         augroup END
-    endif
-endfunction
-
-function! asyncomplete#register_source(info) abort
-    if has_key(s:sources, a:info['name'])
-        return
-    endif
-
-    call s:next_tick_single_exec('clear_active_sources', function('s:clear_active_sources'))
 
-    if has_key(a:info, 'events') && has_key(a:info, 'on_event')
-        execute 'augroup asyncomplete_source_event_' . a:info['name']
-        for l:event in a:info['events']
-            let l:exec =  'if get(b:,"asyncomplete_enable",0) | call s:python_cm_event("' . a:info['name'] . '", "'.l:event.'",asyncomplete#context()) | endif'
-            if type(l:event) == type('')
-                execute 'au ' . l:event . ' * ' . l:exec
-            elseif type(l:event) == type([])
-                execute 'au ' . join(l:event,' ') .' ' .  l:exec
-            endif
-        endfor
-        execute 'augroup end'
+        doautocmd User asyncomplete_setup
```

Comment:
Please define this User event on the top of this file.

```vim
augroup _asyncomplete_silent_
    autocmd!
    autocmd User asyncomplete_setup silent
augroup END
```

Reply:
Fixed. Thanks.